File: backend/src/main/java/com/zzang/chongdae/offeringmember/service/dto/ParticipationResponse.java
package com.zzang.chongdae.offeringmember.service.dto;

import com.zzang.chongdae.offering.domain.OfferingCondition;
import com.zzang.chongdae.offering.domain.OfferingStatus;

public record ParticipationResponse(OfferingCondition offeringCondition, Integer currentCount) {

    public ParticipationResponse(OfferingStatus offeringStatus) {
        this(offeringStatus.decideOfferingCondition(),
                offeringStatus.getCurrentCount()
        );
    }
}

File: android/app/src/main/java/com/zzang/chongdae/remote/dto/response/offering/ParticipationResponse.kt
package com.zzang.chongdae.remote.dto.response.offering

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ParticipationResponse(
    @SerialName("offeringCondition") val offeringCondition: RemoteOfferingStatus,
    @SerialName("currentCount") val currentCount: Int,
)
